src/ca/mcgill/cs/jetuml/diagram/builder/ClassDiagramBuilder.java

```diff
@@ -134,4 +136,196 @@ private PackageNode findContainer( Iterable<Node> pNodes, Point pPoint)
 			}
 		}
 	}
+
+	/*
+	 * Find the pacakage node under the position of the first node in pNodes.
+	 * Return null if there is no such package node for the nodes in pNodes to attach to,
+	 * or the package node is already in the pNodes.
+	 */
+	private PackageNode findPackageToAttach(Iterable<Node> pNodes)
+	{
+	

```

Comment:
Extra blank; "to attach to"; Consider returning an `Optional` instead of null. Also, note that `Diagram` now returns ` List<Node>`, consider using this, see below.